src/samplers/multinomial.jl

```diff
@@ -1,4 +1,4 @@
-function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Float64},
+function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{<:Real},
```

Comment:
AFAICT one has to update the body of the function as well to avoid type stability issues.

Reply:
More concretely, it must not contain any Float64 literals.

Reply:
Isn't is true that the output type of the function can only be `x::AbstractVector{eltype(x)}` ?
I'm adding the type stability tests and it looks ok but I wanted to make sure since I'm pretty new to Julia.

Reply:
There are internal type instabilities which can cause slow downs.

Reply:
I took care of this aspect.
While doing so I noticed that Binomial is restricted to using Float64, so it may be slower/more accurate than necessary.
This also would affect Multinomial.
This is not needed for me, but it can be a good addition in future.